aten/src/ATen/CPUGeneral.cpp

```diff
@@ -0,0 +1,14 @@
+#include <ATen/CPUGeneral.h>
+
+namespace at {
+size_t num_threads = 0;
+std::shared_ptr<tbb::task_scheduler_init> tbbinit;
+void set_num_threads(size_t num_threads_) {
+  if (num_threads == 0) {
+    num_threads = num_threads_;
+    tbbinit = std::make_shared<tbb::task_scheduler_init>(num_threads);
+  }
+}
+
+size_t get_num_threads() { return num_threads; }
```

Comment:
This should probably be part of the ATen context

Reply:
Do you mean part of the at namespace?

Reply:
No I meant the `at::Context` object. That's what we use to hold all other parts of the CPU/CUDA state

Reply:
You probably won't need `shared_ptr` once you move it as well

Reply:
I don't really see how moving this to at::Context improves the API.

`at::set_num_threads()` is both simpler to use and matches the Python API more closely than `at::globalContext().set_num_threads()`.

Putting this in the context seems to imply a false sense of isolation: Multiple task_scheduler_inits will still interfere in separate contexts.

Reply:
I'm just saying that this is where we usually keep all our ATen state, and don't have any other global settings at the moment. If they would somehow interfere then sure, making it global is ok, but I can't see why they would.